src/bgem/bspline/isec_surf_surf.py

```diff
@@ -88,589 +87,466 @@ def _main_curves(surf, axis):
+                                    continue
+
+                            if already_found == False:
+                                lpoint_list.append(point)
+                                point_list.append(point)
+
+                            # TODO: make a method to mark crossings, or comment that it is checked by _already_found
+                            # would be best to have separate small class for that
         return point_list
 
-    @staticmethod
-    def _already_found(crossing, interval_id, curve_id, axis):
+    ##########################
+    # Connecting of the points
+    ##########################
 
-        found = 0
-        ind1 = [curve_id, curve_id]
-        ind2 = [curve_id, curve_id]
-        ind1[1-axis] = interval_id
-        ind2[1-axis] = interval_id + 1
+    def _connect_points(self, point_list1, point_list2):
+        """
+        Builds new data structures in order to connection algorithm may work efficient & call connection algorithm
+        :param point_list1: as the list of the isec_points
+        :param point_list2: as the list of the isec_points
+        :return:
 
-        if np.logical_or(crossing[tuple(ind1)] == 1, crossing[tuple(ind2)] == 1):
-            found = 1
+        TODO: document which attributes of the class are computed/updated
+        """
 
-        return found
+        patch_point1 = self.make_patch_point_list(point_list1, point_list2)
+        patch_point2 = self.make_patch_point_list(point_list2, point_list1)
 
-    # Connecting of the points
+        patch_point = [patch_point1, patch_point2]
+        point_list = [point_list1, point_list2]
 
-    class Patch:
-        def __init__(self, own, other):
-            own_surf_point, other_surf_point = own[0]
-            self.own_surf = own_surf_point.surf
-            self.other_surf = other_surf_point.surf
-            self.main_curve_points = own
-            # Intersection points on main curves of the surface of the patch
-            # Results of get_intersections(own_surf, other_surf).
-            self.other_points = other
-            # Intersection points on main curves of the other surface
+        self._make_point_orderings(point_list, patch_point)
 
+        ## summary of the intersection points (DEBUG)
+        #for point_lists in point_list:
+        #    print(len(point_lists))
+        #    for points in point_lists:
+        #        print(points.xyz)
 
+        ## summary of the curves (DEBUG)
 
-    def _connect_points(self, point_list1, point_list2):
 
-        patch_point_list1, boundary_points1 = self.make_patch_point_list(point_list1, 0)
-        patch_point_list2, boundary_points2 = self.make_patch_point_list(point_list2, 0)
-        patch_point_list1i, boundary_points1i = self.make_patch_point_list(point_list2, 1)
-        patch_point_list2i, boundary_points2i = self.make_patch_point_list(point_list1, 1)
 
-        # print(boundary_points2)
+        print("n_curves=", self.curve_max_id+1)
```

Comment:
Make a function for the usefull debugging informations. Comment out its call in the production code.